### DISCUSSION TITLE
Filament sensor disable/enable

### DISCUSSION THREAD

**Discussion by mvrp350** [Jyers/Marlin#55] [Ideas]
Hi,

thank you for the great firmware update!

I have a smart filament runout sensor installad on my ender 3v2. 
Usually I run my filament throught the sensor, but when I use flexible filament I bypass the sensor with my filament, to prevent false readings.
When bypassing the sensor, I can disable it by using 'M412 S0' in my slicer. This turns out to cause problems, because the program is already started. The sensor is only disabled after the program has started, causing the printer to step into 'M600' filament change.
Is it possible to add a menu item 'enable/disable filament sensor' and also to have the option to enable/disable the sensor in the preheat setting of the material.

Keep up the good work,
mvrp350

**Comment by t3chguy** [comment]
I was planning on doing this, but as the direction the UI seems to be moving is to use existing UI frameworks from within Marlin, that isn't a feature the stock LCD can do in Marlin as far as I can tell so that would not align with that path

**Reply by Jyers** [reply]
The beauty of ExtUI is you can get the best of both worlds. The code can still tie into the marlin UI framework for its calls, but that does not mean its limited to just basic lcd features. Any feature that is hypothetically codable is still totally possible!

**Comment by t3chguy** [comment]
I'll take a crack at this over the weekend, might see if I can get the distance configurable too that way it doesn't have to be something you compile in

**Reply by mvrp350** [reply]
That should also be a great addition!
